Wikipedia/SDImageCache+PromiseKit.swift

```diff
@@ -20,23 +21,25 @@ extension SDImageCache {
             }
         })
 
-        // make sure promise is rejected if the operation is cancelled
-        self.KVOController.observe(diskQueryOperation,
-                                   keyPath: "isCancelled",
-                                   options: NSKeyValueObservingOptions.New)
-        { _, _, change in
-            let value = change[NSKeyValueChangeNewKey] as! NSNumber
-            if value.boolValue {
-                reject(NSError(domain: NSURLErrorDomain, code: NSURLErrorCancelled, userInfo: nil))
+        if let diskQueryOperation = diskQueryOperation {
+            // make sure promise is rejected if the operation is cancelled
+            self.KVOController.observe(diskQueryOperation,
+                keyPath: "isCancelled",
+                options: NSKeyValueObservingOptions.New)
+                { _, _, change in
+                    let value = change[NSKeyValueChangeNewKey] as! NSNumber
+                    if value.boolValue {
+                        reject(NSError(domain: NSURLErrorDomain, code: NSURLErrorCancelled, userInfo: nil))
+                    }
             }
-        }
 
-        // tear down observation when operation finishes
-        self.KVOController.observe(diskQueryOperation,
-                                   keyPath: "isFinished",
-                                   options: NSKeyValueObservingOptions.allZeros)
-        { [weak self] _, object, _ in
-            self?.KVOController.unobserve(object)
+            // tear down observation when operation finishes
+            self.KVOController.observe(diskQueryOperation,
+                keyPath: "isFinished",
+                options: NSKeyValueObservingOptions.allZeros)
+                { [weak self] _, object, _ in
```

Comment:
I know it does in this case, because otherwise we'd get a KVO exception due to the operation being deallocated while observers are attached to it.  I say this, because I ran into it :disappointed:.